cosmos/dbt/graph.py

```diff
@@ -348,7 +378,21 @@ def run_dbt_ls(
         self, dbt_cmd: str, project_path: Path, tmp_dir: Path, env_vars: dict[str, str]
     ) -> dict[str, DbtNode]:
         """Runs dbt ls command and returns the parsed nodes."""
-        ls_command = [dbt_cmd, "ls", "--output", "json"]
+        ls_command = [
```

Comment:
Hey, I noticed that the way dbt ls --output-keys options are supplied has changed across different dbt versions. Here are some examples from various versions:

- [dbt v1.4.2](https://github.com/dbt-labs/dbt-core/blob/v1.4.2/tests/functional/list/test_list.py#L543)
- [dbt v1.3.7](https://github.com/dbt-labs/dbt-core/blob/v1.3.7/tests/functional/list/test_list.py#L543)
- [dbt v1.5.7](https://github.com/dbt-labs/dbt-core/blob/v1.5.7/tests/functional/list/test_list.py#L605)
- [dbt v1.8.3](https://github.com/dbt-labs/dbt-core/blob/v1.8.3/tests/functional/list/test_list.py#L741)

For version 1.5.x and the latest expected like below 
```bash
ls_command = [
  ....
  "--output-keys",
  "name",
  "unique_id"
]
```
And for 1.4.x and older it is like 

```
ls_command = [
  ...
  "--output-keys",
  "name, unique_id",
]
```

To handle both cases, I can see 2 option

1. Run the dbt version command, parse the version number, and then construct the appropriate dbt command based on that version. The downside is that this might be a bit slow and could introduce bugs if the output format of dbt -v changes.
2. Introduce a config variable, TEST_SOURCE_GT_V14, to determine the command construction. While this approach might seem a bit unconventional from a user experience side, it would be relatively bug-free and faster.

I'd appreciate any thoughts or suggestions you might have on this.

Reply:
@pankajastro, there is a third approach:

a) Try to run the following (which would work if `dbt-core>=1.5`):
```
ls_command = [ .... "--output-keys", "name", "unique_id" ]
```

b) If it fails (is there some specific message in the command output we could check, it would be great), we then try to run using the `dbt-core<=1.4` approach:
```
ls_command = [ ... "--output-keys", "name, unique_id", ]
```

The advantage of this approach is that it would not incur additional costs to end-users already using the latest versions of dbt. In contrast, it would incur one more dbt subprocess call to users who are using older versions of dbt-core, without the need for any additional configuration.

The cost for both of these is better than approach (1) because it would not be running subprocess twice every time.  Both of them will have performance improvements once we implement:
https://github.com/astronomer/astronomer-cosmos/issues/865

The challenge with approach (2) is to force users to set yet another configuration.

Approch (4), which may be cleaner on the Cosmos side, is to describe in the docs that Source nodes are only supported with dbt-core>=1.5.

Reply:
@tatiana & @pankajastro discussed this further and we feel with approach (4), we however also need to capture the specific error for invalid command and raise relevant warnings/logs

Reply:
Yes, the 4th option seems reasonable to me. This is what I believe we should proceed with.

1. In the current implementation, the default behavior of source_rendering_behavior is set to SourceRenderingBehavior.ALL. I propose changing this default behavior to SourceRenderingBehavior.NONE. This adjustment ensures that the default behavior remains consistent with the old DAG rendering behavior, providing no change for users by default.
https://github.com/astronomer/astronomer-cosmos/blob/d788c040e87e9d11d06f3cf605c30683ca5ae4c3/cosmos/config.py#L80
```python
source_rendering_behavior: SourceRenderingBehavior = SourceRenderingBehavior.NONE 
```
2. If source node rendering is SourceRenderingBehavior.NONE then keep dbt ls command as it was before this PR otherwise change the command which work for dbt-core >= 1.5.0.
Something like
```
ls_command = []

if self.render_config.source_rendering_behavior != SourceRenderingBehavior.NONE:
    ls_command = [
        dbt_cmd,
        "ls",
        "--output",
        "json",
        "--output-keys",
        "name",
        "unique_id",
        "resource_type",
        "depends_on",
        "original_file_path",
        "tags",
        "config",
        "freshness"
    ]
else:
    ls_command = [dbt_cmd, "ls", "--output", "json"]

ls_args = self.dbt_ls_args
ls_command.extend(self.local_flags)
ls_command.extend(ls_args)

try:
    stdout = run_command(ls_command, tmp_dir, env_vars)
except CosmosLoadDbtException as e:
    if self.render_config.source_rendering_behavior != SourceRenderingBehavior.NONE:
        logger.warning(f"cosmos support source node rendering for dbt-core > 1.5.0 {e}")
    else:
        raise e
```
https://github.com/astronomer/astronomer-cosmos/blob/d788c040e87e9d11d06f3cf605c30683ca5ae4c3/cosmos/dbt/graph.py#L381
3. Add `source_rendering_behavior` into the integration tests that run for dbt-core >= 1.5.0
4. Document that this feature is available only dbt-core >=1.5.0